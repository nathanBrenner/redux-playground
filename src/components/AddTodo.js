import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { addTodo } from "../actions";

let AddTodo = ({ dispatch }) => {
  let input;
  const comp = (
    <div>
      <input
        ref={node => {
          input = node;
        }}
      />
      <button
        onClick={() => {
          dispatch(addTodo(input.value));
          input.value = "";
        }}
      >
        Add Todo
      </button>
    </div>
  );
  console.log(comp);
  return comp;
};

AddTodo.propTypes = {
  dispatch: PropTypes.func.isRequired
};

export default connect()(AddTodo);
