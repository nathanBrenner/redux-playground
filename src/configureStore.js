import todoApp from "./reducers";
import { createStore, applyMiddleware } from "redux";
import promise from "redux-promise";
import { createLogger } from "redux-logger";

const configureStore = () => {
  const middlewares = [promise];

  if (process.env.NODE_ENV !== "production") {
    middlewares.push(createLogger());
  }

  const store = createStore(
    todoApp,
    window.__REDUX_DEVTOOLS_EXTENSION__ &&
      window.__REDUX_DEVTOOLS_EXTENSION__(),
    applyMiddleware(...middlewares)
  );
  //     window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  return store;
};

export default configureStore;
