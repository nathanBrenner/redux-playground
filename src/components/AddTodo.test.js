import React from "react";
import AddTodo from "./AddTodo";
import configureStore from "../configureStore";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("AddTodo", () => {
  test("renders without crashing", () => {
    const store = configureStore();
    shallow(<AddTodo store={store} />);
  });
  test("snapshot", () => {
    const store = configureStore();
    const component = renderer.create(<AddTodo store={store} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
