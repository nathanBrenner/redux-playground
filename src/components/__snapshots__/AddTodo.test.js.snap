// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo snapshot 1`] = `
<div>
  <input />
  <button
    onClick={[Function]}
  >
    Add Todo
  </button>
</div>
`;
